src/Idris/Desugar.idr

```diff
@@ -277,6 +272,21 @@ mutual
       = pure $ IMustUnify fc UserDotted !(desugarB side ps x)
   desugarB side ps (PImplicit fc) = pure $ Implicit fc True
   desugarB side ps (PInfer fc) = pure $ Implicit fc False
+  desugarB side ps (PString fc strs)
+      = case !fromStringName of
+             Nothing => pure $ !(desugurStr strs)
+             Just f => pure $ IApp fc (IVar fc f)
+                                      !(desugurStr strs)
+    where
+      toRawImp : PStr -> Core RawImp
+      toRawImp (StrLiteral fc s) = pure $ IPrimVal fc (Str s)
+      toRawImp (StrInterp fc tm) = desugarB side ps tm
+      concat : RawImp -> RawImp -> RawImp
+      concat a b = IApp (getFC a) (IApp (getFC b) (IVar (getFC b) (UN "++")) a) b
```

Comment:
Should we instead call `fastConcat`?

Reply:
I don't know. Is there any case where `++` is needed? If no, should `++` be an alias for `fastConcat`?

Reply:
`fastConcat` takes a whole `List` of strings and only allocates one big
string that contains the concatenation of them all. `foldr (++)` will
leads to the allocation of many intermerdiate strings.

`(++)` still has its uses when you're only putting 2 strings together.

Reply:
I see. Another question is that `++` isn't in prelude. So is it a good choice to include ffi in the prelude?

Reply:
Ah, I see. I guess it would make sense to add this as a primitive and insist that
each backend should implement it. But we can leave that as future work, in a separate PR.